src/modules/dashboard/sagas/utils/safeHelpers.ts

```diff
@@ -280,6 +281,92 @@ export const getTransferFundsData = async (
   ]);
 };
 
+const getParsedJSONOrArrayFromString = (
+  parameter: string,
+): (string | number)[] | null => {
+  try {
+    const arrayResult = JSON.parse(parameter);
+    return arrayResult.map((value) => {
+      if (Number.isInteger(value)) {
+        return new BigNumber(value).toString();
+      }
+      return value;
+    });
+  } catch (err) {
+    return null;
+  }
+};
+
+const isArrayParameter = (parameter: string): boolean =>
+  /(\[\d*])+$/.test(parameter);
+
+const extractMethodArgs = (
+  signature: string,
+  transaction: Record<string, any>,
+) => ({ name = '' }) => {
+  if (isArrayParameter(transaction[name || signature])) {
+    return getParsedJSONOrArrayFromString(transaction[name || signature]);
+  }
+  return transaction[name || signature];
+};
+
+export const getContractInteractionData = async (
+  zodiacBridgeModule: Contract,
+  safe: Omit<ColonySafe, 'safeName'>,
+  transaction: SafeTransaction,
+) => {
+  const safeAddress = onLocalDevEnvironment
+    ? LOCAL_SAFE_ADDRESS
+    : safe.contractAddress;
+  const contractAddress = onLocalDevEnvironment
+    ? LOCAL_SAFE_TOKEN_ADDRESS
+    : transaction.contract.profile.walletAddress;
+  let abi: string;
+  let contractFunction: string;
+
+  if (!safeAddress) {
+    throw new Error('LOCAL_SAFE_ADDRESS not set in .env.');
+  }
+
+  if (!contractAddress) {
+    throw new Error('LOCAL_SAFE_TOKEN_ADDRESS not set in .env.');
+  }
+
+  if (onLocalDevEnvironment) {
+    const TokenInterface = await getTokenInterface(safe, contractAddress);
+    abi = JSON.stringify(TokenInterface.abi);
+    contractFunction = 'transferFrom';
+  } else {
+    abi = transaction.abi;
+    contractFunction = transaction.contractFunction;
+  }
+
+  const getData = () => {
+    const foreignProvider = getForeignProvider(safe);
+    const contract = new ethers.Contract(contractAddress, abi, foreignProvider);
+    const { inputs, name = '', signature } = contract.interface.functions[
+      contractFunction
+    ];
+
+    const transactionValues = onLocalDevEnvironment
+      ? { ...transaction, src: safeAddress, dst: AddressZero, wad: 1 }
```

Comment:
Perhaps worth a comment here that `src`, `dst` and `wad` are the inputs to the `transferFrom` function